stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -778,26 +787,42 @@ internal constructor(
     //endregion
 
     public fun disconnectSocket() {
-        socket.disconnect()
+        if (ToggleService.isSocketExperimental()) {
+            chatSocketExperimental.disconnect(DisconnectCause.UnrecoverableError(null))
```

Comment:
Actually, in existing implementation, we pass DisconnectCause `DisconnectCause.UnrecoverableError(null)`. If we change, it would be a breaking change. Do you think we should do that?

Reply:
:thinking: 
You are using the new implementation here, how it can be a breaking change?

Reply:
In case of Disconnection, we call `SocketListener.onDisconnected(disconnectCause)`. If previously for the same behavior, we invoke `onDisconnected(DisconnectCause.UnrecoverablerError)` and now if we invoke `onDisconnected(DisconnectCause.ConnectionReleased)`, will it not create problems for users who are listening to this and doing some action based on the type of `DisconnectCause`?

Reply:
It is a sealed class, if they are observing changes in it, they must handle this `DisconnectCause`.
And just into this method the `DisconnectCause` is `ConnectionReleased`, The SDK was asked to disconnect, so no `UnrecoverableError`